pkg/controllers/awsloadbalancercontroller/deployment.go

```diff
@@ -15,16 +17,20 @@ import (
 
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/operator-framework/operator-lib/proxy"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 
 	albo "github.com/openshift/aws-load-balancer-operator/api/v1alpha1"
 )
 
 const (
-	appLabelName    = "app.kubernetes.io/name"
-	appName         = "aws-load-balancer-operator"
-	appInstanceName = "app.kubernetes.io/instance"
+	appLabelName        = "app.kubernetes.io/name"
+	appName             = "aws-load-balancer-operator"
+	appInstanceName     = "app.kubernetes.io/instance"
+	trustedCAAnnotation = "networking.olm.openshift.io/trusted-ca-configmap-hash"
```

Comment:
Can you make this work without adding an annotation?  If not, why not?

Reply:
No. The annotation is needed in the template pod spec of the managed controller to re-trigger the deployment in case the contents of the trusted CA configmap changed. The annotation contains the hash of the trusted CA configmap: changed configmap contents => changed hash => changed template podspec => redeployment of the aws load balancer controller.

Reply:
We use a different idiom to compute changes, everywhere else in OpenShift.  We compare the previous and new, and don't need to add annotations.  I would prefer if you follow that idiom.

Reply:
I see what you mean. Yes, all the operators compare the new and the desired states. The difference here is that the trusted CA configmap is not part of the desired state. The operator takes it as an opaque object. However the operator's job is to ensure this configmap is consumed by the managed aws lb controller's deployment. That is done by having the volume and the volumeMount as a part of the desired state (`pkg/controllers/awsloadbalancercontroller/deployment.go` line 231).
The annotation though is just a way to let the aws-load-balancer-controller reload CA certs from the host (container). The annotation would not be necessary if the aws-load-balancer-controller could [fsnotify](https://pkg.go.dev/github.com/fsnotify/fsnotify) on the system certs.

Reply:
@alebedev87 , can you add godoc to `trustedCAAnnotation` to explain that its purpose is to force the controller to restart in order to reload the trusted CA when it changes?

Reply:
@Miciah : comment added. Also the comment in `controller.go#BuildManagedController` was updated  to explain better what happens after the reconciliation gets triggered from the trusted CA configmap events.